airflow/contrib/operators/dynamodb_to_s3.py

```diff
@@ -0,0 +1,146 @@
+
+    To parallelize the replication, users can create multiple tasks of DynamoDBToS3Operator.
+    For instance to replicate with parallelism of 2, create two tasks like:
+
+    .. code-block::
+
+        op1 = DynamoDBToS3Operator(
+            task_id='replicator-1',
+            dynamodb_table_name='hello',
+            dynamodb_scan_kwargs={
+                'TotalSegments': 2,
+                'Segment': 0,
+            },
+            ...
+        )
+
+        op2 = DynamoDBToS3Operator(
+            task_id='replicator-2',
+            dynamodb_table_name='hello',
+            dynamodb_scan_kwargs={
+                'TotalSegments': 2,
+                'Segment': 1,
+            },
+            ...
+        )
+
+    :param dynamodb_table_name: Dynamodb table to replicate data from
+    :param s3_bucket_name: S3 bucket to replicate data to
+    :param file_size: Flush file to s3 if file size >= file_size
+    :param dynamodb_scan_kwargs: kwargs pass to <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Table.scan>  # noqa: E501 pylint: disable=line-too-long
+    :param s3_key_prefix: Prefix of s3 object key
+    :param process_func: How we transforms a dynamodb item to bytes. By default we dump the json
+    """
+
+    def __init__(self,
+                 dynamodb_table_name: str,
+                 s3_bucket_name: str,
+                 file_size: int,
+                 dynamodb_scan_kwargs: Optional[Dict[str, Any]] = None,
+                 s3_key_prefix: str = '',
+                 process_func: Callable[[Dict[str, Any]], bytes] = _convert_item_to_json_bytes,
+                 *args, **kwargs):
+        super(DynamoDBToS3Operator, self).__init__(*args, **kwargs)
+        self.file_size = file_size
+        self.process_func = process_func
+        self.dynamodb_table_name = dynamodb_table_name
+        self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
+        self.s3_bucket_name = s3_bucket_name
+        self.s3_key_prefix = s3_key_prefix
+
+    def execute(self, context):
+        table = AwsDynamoDBHook().get_conn().Table(self.dynamodb_table_name)
+        scan_kwargs = copy(self.dynamodb_scan_kwargs) if self.dynamodb_scan_kwargs else {}
+        err = None
+        f = NamedTemporaryFile()
+        try:
+            f = self._scan_dynamodb_and_upload_to_s3(f, scan_kwargs, table)
+        except Exception as e:
+            err = e
+            raise e
+        finally:
+            if err is None:
+                _upload_file_to_s3(f, self.s3_bucket_name, self.s3_key_prefix)
+            f.close()
+
+    def _scan_dynamodb_and_upload_to_s3(self, f, scan_kwargs, table):
+        while True:
+            response = table.scan(**scan_kwargs)
+            items = response['Items']
+            for item in items:
+                f.write(self.process_func(item))
+
+            if 'LastEvaluatedKey' not in response:
+                # no more items to scan
+                break
```

Comment:
IMO this if block with `break` should be in the end of `while` block?

Reply:
The reason I put the here is that there is that if the file size reaches threshold in the last run. It only flush to S3 once in https://github.com/apache/airflow/pull/5663/files#diff-98426c5c739f27ea6aa369d3be536186R124.

Otherwise if we move the `break` in the end of `while` block, it got flush twice:
1. https://github.com/apache/airflow/pull/5663/files#diff-98426c5c739f27ea6aa369d3be536186R143
2. https://github.com/apache/airflow/pull/5663/files#diff-98426c5c739f27ea6aa369d3be536186R124